diff --git a/source/cmake/build/build.settings.cmake b/source/cmake/build/build.settings.cmake
index 1f73699632b..b0efe0745b3 100644
--- a/source/cmake/build/build.settings.cmake
+++ b/source/cmake/build/build.settings.cmake
@@ -52,7 +52,6 @@ if( ${COMPILER} STREQUAL "gcc" )
 			-Wall
 			-Wextra
 			-pedantic
-			-Werror  # REMOVE FOR RELEASE
 			-Wno-long-long
 			-Wno-strict-aliasing
 	)
diff --git a/source/cmake/build_cxx11_omp/CMakeLists.txt b/source/cmake/build_cxx11_omp/CMakeLists.txt
index 23082655a18..622c1845ae5 100644
--- a/source/cmake/build_cxx11_omp/CMakeLists.txt
+++ b/source/cmake/build_cxx11_omp/CMakeLists.txt
@@ -16,7 +16,7 @@ SET(LINK_RULES shared)
 INCLUDE(../build/compiler.cmake)
 INCLUDE(../build/build.pre.cmake)
 INCLUDE(../build/${LINK_RULES}.build.cmake)
-# INCLUDE(../build/build.settings.cmake)
+INCLUDE(../build/build.settings.cmake)

 ADD_DEFINITIONS(-DUSEOMP)
 ADD_DEFINITIONS(-DUSE_OPENMP)
@@ -32,6 +32,11 @@ SET_SOURCE_FILES_PROPERTIES( ../../src/protocols/cryst/spacegroup.cc PROPERTIES
 SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fopenmp -ffor-scope -std=c++11 -pipe -march=core2 -mtune=generic -O3 -ffast-math -fno-finite-math-only -funroll-loops -finline-functions -finline-limit=20000 -s")
 SET(CMAKE_SHARED_LINKER_FLAGS "-lgomp")
 SET(CMAKE_EXE_LINKER_FLAGS "-lgomp  -L/usr/local/lib")
+
+if(DEFINED CONDA_LIB_PREFIX)
+  add_link_options("-L${CONDA_LIB_PREFIX}")
+endif()
+
 include_directories("/usr/local/include")


@@ -41,3 +46,5 @@ INCLUDE(../build/build.post.cmake)
 INCLUDE( ../build/apps.all.cmake )
 INCLUDE( ../build/pilot_apps.all.cmake )

+############ Installation rules
+INCLUDE( ../build/install.cmake )
