--- Rosetta/source/src/core/io/silent/ProteinSilentStruct.hh.orig	2021-06-06 08:12:24.000000000 -0700
+++ Rosetta/source/src/core/io/silent/ProteinSilentStruct.hh	2021-06-06 08:12:28.000000000 -0700
@@ -72,7 +72,7 @@
 		resize( nres_in );
 		//fold_tree_ = core::kinematics::FoldTree();
 		bJumps_use_IntraResStub_ = false;
-		symminfo_ = new core::conformation::symmetry::SymmetryInfo();
+		symminfo_ = core::conformation::symmetry::SymmetryInfoOP( new core::conformation::symmetry::SymmetryInfo() );
 		symminfo_->set_use_symmetry(false);
 	}

--- Rosetta/source/cmake/build/build.settings.cmake.orig	2021-06-06 08:11:36.000000000 -0700
+++ Rosetta/source/cmake/build/build.settings.cmake	2021-06-06 16:57:17.000000000 -0700
@@ -52,7 +52,6 @@
 			-Wall
 			-Wextra
 			-pedantic
-			-Werror  # REMOVE FOR RELEASE
 			-Wno-long-long
 			-Wno-strict-aliasing
 	)
@@ -245,7 +244,6 @@
 			-Wextra
 			-pedantic
 			#-Weverything
-			-Werror # REMOVE FOR RELEASE
 			-Wno-long-long
 			-Wno-strict-aliasing
 			#-Wno-documentation
@@ -278,6 +276,7 @@
 if(APPLE AND ${COMPILER} STREQUAL "clang")
     list( REMOVE_ITEM compile
 	-stdlib=libstdc++
+	-I\ src/platform/linux
     )
     list( REMOVE_ITEM shlink
 	-stdlib=libstdc++
--- Rosetta/source/cmake/build_cxx11_omp/CMakeLists.txt.orig	2021-06-06 08:11:56.000000000 -0700
+++ Rosetta/source/cmake/build_cxx11_omp/CMakeLists.txt	2021-06-06 08:18:06.000000000 -0700
@@ -9,14 +9,14 @@

 ############ Build Type, Mode, Link Rules, Compiler, Extras
 SET(MODE release)
-SET(COMPILER gcc)
 SET(LINK_RULES shared)
+SET(CMAKE_EXPORT_COMPILE_COMMANDS 1)

 ############ Build Rules
 INCLUDE(../build/compiler.cmake)
 INCLUDE(../build/build.pre.cmake)
 INCLUDE(../build/${LINK_RULES}.build.cmake)
-# INCLUDE(../build/build.settings.cmake)
+INCLUDE(../build/build.settings.cmake)

 ADD_DEFINITIONS(-DUSEOMP)
 ADD_DEFINITIONS(-DUSE_OPENMP)
@@ -30,8 +30,16 @@
 SET_SOURCE_FILES_PROPERTIES( ../../src/protocols/cryst/spacegroup.cc PROPERTIES COMPILE_FLAGS -O0 )

 SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fopenmp -ffor-scope -std=c++11 -pipe -march=core2 -mtune=generic -O3 -ffast-math -fno-finite-math-only -funroll-loops -finline-functions -finline-limit=20000 -s")
-SET(CMAKE_SHARED_LINKER_FLAGS "-lgomp")
-SET(CMAKE_EXE_LINKER_FLAGS "-lgomp  -L/usr/local/lib")
+
+if( ${COMPILER} STREQUAL "gcc" )
+  SET(CMAKE_SHARED_LINKER_FLAGS "-lgomp")
+  SET(CMAKE_EXE_LINKER_FLAGS "-lgomp  -L/usr/local/lib")
+endif()
+
+if(DEFINED CONDA_LIB_PREFIX)
+  add_link_options("-L${CONDA_LIB_PREFIX}")
+endif()
+
 include_directories("/usr/local/include")


@@ -41,3 +49,5 @@
 INCLUDE( ../build/apps.all.cmake )
 INCLUDE( ../build/pilot_apps.all.cmake )

+############ Installation rules
+INCLUDE( ../build/install.cmake )
